include/nbl/video/IDescriptorPool.h

```diff
@@ -29,12 +33,156 @@ class NBL_API IDescriptorPool : public core::IReferenceCounted, public IBackendO
+        {
+            memset(m_maxDescriptorCount, 0, asset::EDT_COUNT * sizeof(uint32_t));
+
+            for (auto i = 0; i < poolSizeCount; ++i)
+                m_maxDescriptorCount[poolSizes[i].type]++;
+
+            for (auto i = 0; i < asset::EDT_COUNT; ++i)
+            {
+                if (m_maxDescriptorCount[i] > 0)
+                {
+                    if (m_flags & ECF_FREE_DESCRIPTOR_SET_BIT)
+                    {
+                        m_generalAllocatorReservedSpace[i] = _NBL_ALIGNED_MALLOC(core::GeneralpurposeAddressAllocator<uint32_t>::reserved_size(alignof(core::smart_refctd_ptr<asset::IDescriptor>), m_maxDescriptorCount[i] * sizeof(void*), 1u), _NBL_SIMD_ALIGNMENT);
+                        m_generalAllocators[i] = core::GeneralpurposeAddressAllocator<uint32_t>(m_generalAllocatorReservedSpace[i], 0u, 0u, 1u, m_maxDescriptorCount[i] * sizeof(void*), 1u);
+                    }
+                    else
+                    {
+                        m_linearAllocators[i] = core::LinearAddressAllocator<uint32_t>(nullptr, 0u, 0u, 1u, m_maxDescriptorCount[i]*sizeof(void*));
+                    }
+                }
+            }
 
-        uint32_t getCapacity() const { return m_maxSets; }
+            // Initialize the storages.
+            m_combinedImageSamplerStorage = std::make_unique<StorageTrivializer<SCombinedImageSampler>[]>(m_maxDescriptorCount[asset::EDT_COMBINED_IMAGE_SAMPLER]);
+            m_storageImageStorage = std::make_unique<StorageTrivializer<core::smart_refctd_ptr<IGPUImageView>>[]>(m_maxDescriptorCount[asset::EDT_STORAGE_IMAGE] + m_maxDescriptorCount[asset::EDT_INPUT_ATTACHMENT]);
+            m_UBO_SSBOStorage = std::make_unique<StorageTrivializer<core::smart_refctd_ptr<IGPUBuffer>>[]>(m_maxDescriptorCount[asset::EDT_UNIFORM_BUFFER] + m_maxDescriptorCount[asset::EDT_STORAGE_BUFFER] + m_maxDescriptorCount[asset::EDT_UNIFORM_BUFFER_DYNAMIC] + m_maxDescriptorCount[asset::EDT_STORAGE_BUFFER_DYNAMIC]);
+            m_UTB_STBStorage = std::make_unique<StorageTrivializer<core::smart_refctd_ptr<IGPUBufferView>>[]>(m_maxDescriptorCount[asset::EDT_UNIFORM_TEXEL_BUFFER] + m_maxDescriptorCount[asset::EDT_STORAGE_BUFFER_DYNAMIC]);
+            m_accelerationStructureStorage = std::make_unique<StorageTrivializer<core::smart_refctd_ptr<IGPUAccelerationStructure>>[]>(m_maxDescriptorCount[asset::EDT_ACCELERATION_STRUCTURE]);
+        }
+
+        ~IDescriptorPool()
+        {
+            for (auto i = 0; i < asset::EDT_COUNT; ++i)
+            {
+                if ((m_flags & IDescriptorPool::ECF_FREE_DESCRIPTOR_SET_BIT) && (m_maxDescriptorCount[i] > 0))
+                    _NBL_ALIGNED_FREE(m_generalAllocatorReservedSpace[i]);
+            }
```

Comment:
make `m_generalAllocatorReservedSpace` a `std::unique_ptr<uint8_t[]>` so you don't need to remember (`new` allocates aligned to 16 bytes/SSE anyway)